### DISCUSSION TITLE
Meaning of -1.000 in AP and AR

### DISCUSSION THREAD

**Discussion by AnhMinhTran** [rwightman/efficientdet-pytorch#219] [Q&A]
As I understand, the AP amd AR calculation are as of follow:
Precision = truepositive/ (truepositive + falsepositive)
Recall = truepositive/ (truepositive + falsenegative)

Which is what I found in evaluation dir
```
precision = cum_true_positives.astype(float) / (cum_true_positives + cum_false_positives)
    recall = cum_true_positives.astype(float) / num_gt
```

![Screenshot_20210504-012541_Gallery](https://user-images.githubusercontent.com/29029296/116896558-ba342f80-ac77-11eb-9c74-711e456d5f89.jpg)

So if my AR and AP is consistently equal -1.000, the model doesn't learn and uodate any weights throughout the epochs?

**Comment by rwightman** [comment]
@AnhMinhTran I don't think I've ever seen -1 and I've run a lot of models using this code and elsewhere through the coco evaluator, there must be something wrong with the predictions... it's generally 0. if it's not learning.

**Reply by AnhMinhTran** [reply]
I have figured out the issue, its due to labeling categories in the JSON file. The JSON I got the annotation starts with 0 index, which caused this issue
